test/index.test.ts

```diff
@@ -15,91 +15,187 @@
+      '.container_, .full-height .image_.full_-width_ {}'
+    );
+  });
 });
 
-it('renames excluding provided classes', async () => {
-  const input = await read('default.css');
-  const expectedOutput = await read('default.closure.no-image.css');
+describe('with strategy "minimal"', () => {
+  it('maps parts to the shortest possible strings', async () => {
+    assertPostcss(
+      await run(INPUT, { strategy: 'minimal' }),
+      '.a, .b-c .d.b-e {}'
+    );
+  });
 
-  assertPostcss(await run(input, { renamingType: 'CLOSURE', excludedClassesFromRenaming: ['image'] }), expectedOutput);
-});
+  it('adds a prefix after minimizing', async () => {
+    assertPostcss(
+      await run(INPUT, { strategy: 'minimal', prefix: 'pf-' }),
+      '.pf-a, .pf-b-c .pf-d.pf-b-e {}'
+    );
+  });
 
-it.each([
-  ['CLOSURE_COMPILED_BY_WHOLE', 'js'],
-  ['CLOSURE_COMPILED_SPLIT_HYPHENS', 'js'],
-  ['CLOSURE_COMPILED', 'js'],
-  ['CLOSURE_UNCOMPILED', 'js'],
-  ['JSCOMP_VARIABLE_MAP', 'map'],
-  ['JSON', 'json'],
-  ['PROPERTIES', 'properties'],
-])('outputs %s renaming map', async (format, extension) => {
-  const outputName = `renaming_map.${format}.${extension}`;
-  const input = await read('default.css');
-  const expectedOutput = await read(outputName);
-
-  mockFs({ 'temp/': {} });
-  await run(input, {
-    renamingType: 'CLOSURE',
-    outputRenamingMap: `temp/${outputName}`,
-    outputRenamingMapFormat: format as keyof typeof OutputRenamingMapFormat,
+  it('maps original names to minimized names', async () => {
+    await assertMapEquals(
+      INPUT,
+      {
+        container: 'a',
+        full: 'b',
+        height: 'c',
+        image: 'd',
+        width: 'e',
+      },
+      { strategy: 'minimal' }
+    );
+  });
+
+  it("doesn't map excluded names", async () => {
+    assertPostcss(
+      await run(INPUT, { strategy: 'minimal', except: ['full-height'] }),
+      '.a, .full-height .b.c-d {}'
+    );
   });
-  const output = (await fs.readFile(`temp/${outputName}`)).toString();
-  mockFs.restore();
 
-  expect(output).toEqual(expectedOutput);
+  it("doesn't produce a name that would be excluded", async () => {
+    assertPostcss(
+      await run(INPUT, { strategy: 'minimal', except: ['b'] }),
+      '.a, .c-d .e.c-f {}'
+    );
+  });
+
+  describe('toShortName()', () => {
+    it('produces the right results around the two-character boundary', () => {
+      expect(toShortName(52)).toEqual('_');
+      expect(toShortName(53)).toEqual('aa');
+      expect(toShortName(54)).toEqual('ba');
+      expect(toShortName(55)).toEqual('ca');
+    });
+
+    it('produces the right results around the three-character boundary', () => {
+      expect(toShortName(3390)).toEqual('Z_');
+      expect(toShortName(3391)).toEqual('__');
+      expect(toShortName(3392)).toEqual('aaa');
+      expect(toShortName(3393)).toEqual('baa');
+      expect(toShortName(3394)).toEqual('caa');
+    });
+  });
```

Comment:
Nit: Could MinimalSubstitutionMapTest#testToShortString be worth keeping? Maybe in another PR.